integration-tests/security/src/test/java/io/quarkiverse/openapi/generator/it/security/OpenWeatherTest.java

```diff
@@ -32,15 +35,45 @@ public class OpenWeatherTest {
 
     @RestClient
     @Inject
-    CurrentWeatherDataApi weatherApi;
+    CurrentWeatherDataV2Api weatherApiv2;
 
     @Test
     public void testGetWeatherByLatLon() {
-        final Model200 model = weatherApi.currentWeatherData("", "", "10", "-10", "", "", "", "");
+        final Model200 model = weatherApiv2.currentWeatherData("", "", "10", "-10", "", "", "", "");
         assertEquals("Nowhere", model.getName());
         assertNotNull(weatherUrl);
         openWeatherServer.verify(WireMock.getRequestedFor(
                 WireMock.urlEqualTo("/data/2.5/weather?q=&id=&lat=10&lon=-10&zip=&units=&lang=&mode=&appid=" + apiKey)));
     }
 
+    @Test
+    public void testClientHeadersFactory_defaultClientHeadersFactory() throws ClassNotFoundException {
+        Class currentWeatherDataV2Api = this.getClass().getClassLoader()
+                .loadClass("org.acme.openapi.weather.v2.api.CurrentWeatherDataV2Api");
+        assertTrue(currentWeatherDataV2Api.isAnnotationPresent(RegisterClientHeaders.class));
+        RegisterClientHeaders annotation = (RegisterClientHeaders) currentWeatherDataV2Api
+                .getAnnotation(RegisterClientHeaders.class);
+        assertEquals(AuthenticationPropagationHeadersFactory.class, annotation.value());
+    }
+
+    @Test
+    public void testClientHeadersFactory_noneClientHeadersFactory() throws ClassNotFoundException {
+        Class currentWeatherDataV3Api = this.getClass().getClassLoader()
+                .loadClass("org.acme.openapi.weather.v3.api.CurrentWeatherDataV3Api");
+        assertTrue(currentWeatherDataV3Api.isAnnotationPresent(RegisterClientHeaders.class));
+        RegisterClientHeaders annotation = (RegisterClientHeaders) currentWeatherDataV3Api
```

Comment:
```suggestion
        RegisterClientHeaders annotation = currentWeatherDataV3Api.getAnnotation(RegisterClientHeaders.class);
```

Reply:
It will compile if you declare `currentWeatherDataV3Api` as `Class<?>`.

Reply:
ok fixed